### PR TITLE
Support FreeBSD and Darwin

### LINKED ISSUE #238: FreeBSD Support
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: all
- Ruby: all
- Distribution: FreeBSD (any)
- Module version: 5.1.1

## How to reproduce (e.g Puppet code you use)
n/a

## What are you seeing
```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Function Call, Could not find template 'snmp/snmpd.sysconfig-FreeBSD.erb' (file: /etc/puppetlabs/code/environments/pdk/modules/snmp/manifests/init.pp, line: 439, column: 16) on node archive4.int.hpwren.ucsd.edu
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
```

## What behaviour did you expect instead
Working on FreeBSD.

## Any additional information you'd like to impart
It'd be nice to be able to be able to override the various templates for the individual files. They are currently hard-coded in init.pp, making them difficult to change out without forking the module.

### PR DESCRIPTION
**Pull request by geoffdavis**
#### This Pull Request (PR) fixes the following issues
Fixes #238

### CODE REVIEW

**Review by kenyon — APPROVED** [review]

File: manifests/init.pp [hunk 1]
```diff
@@ -440,19 +472,21 @@
...
-  file { 'snmpd.sysconfig':
-    ensure  => $file_ensure,
-    path    => $sysconfig,
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0644',
-    content => template($template_snmpd_sysconfig),
-    require => Package['snmpd'],
-    notify  => Service['snmpd'],
+  unless $facts['os']['family'] in ['FreeBSD', 'Darwin'] {
```
Comment (ghoneycutt) [inline manifests/init.pp:479]: nit: if not is easier to understand than unless

File: manifests/init.pp [hunk 2]
```diff
@@ -494,12 +528,13 @@
...
-      require    => [
-        File['var-net-snmp'],
-        Package['snmptrapd'],
-      ],
+      require    => File['var-net-snmp'],
       subscribe  => File['snmptrapd.conf'],
     }
+
+    if $manage_packages {
+      Package['snmptrapd'] -> Service['snmptrapd']
```
Comment (ghoneycutt) [inline manifests/init.pp:536]: This would be easier to understand if the package just had a `before` attribute.

File: manifests/init.pp (hunk 1 above)
Comment (geoffdavis) [reply manifests/init.pp:479]: I disagree here, "if !" seems less succinct than "unless", as the "!" is often overlooked when scanning through code. That said, if this is what is blocking the merge 6 months on, I'm happy to change it.

File: manifests/init.pp (hunk 2 above)
Comment (geoffdavis) [reply manifests/init.pp:536]: While I agree, I was following the style of the existing code, rather than turning this PR into a gigantic refactor.

**Comment by geoffdavis** [comment]
Thanks @ghoneycutt !

**Pull request merged**